doc/user-guide.md

```diff
@@ -249,6 +249,90 @@ return reconcile.Result{RequeueAfter: time.Second*5}, nil
 
 **Note:** Returning `Result` with `RequeueAfter` set is how you can periodically reconcile a CR.
 
+#### Reconcile Result Use Cases
+**The following are possible reconcile loop return options.**
+
+#### 1. With the error:
+
+If an error is encountered during processing the appropriate return option is to return an error.
+This results in the reconcile loop being re-triggered to run again.
+
+Usage
+```Go
+return reconcile.Result{}, err
+```
+
+Example:
+
+In the [`memcached_controller.go`][memcached_controller] example below a `reconcile.Result{}, err` is used when there is an error reading the object.
+As a result the request is requeued for another try.
+```Go
+// Fetch the Memcached instance
+memcached := &cachev1alpha1.Memcached{}
+err := r.client.Get(context.TODO(), request.NamespacedName, memcached)
+if err != nil {
+	if errors.IsNotFound(err) {
+		...
+  }
+  // Error reading the object - requeue the request.
+	reqLogger.Error(err, "Failed to get Memcached")
+	return reconcile.Result{}, err
+}
+```
+
+#### 2. Without an error:
+
+There are several situations where although no error occured, the reconcile loop should signify
+during its return that it needs to run again.
+
+Usage
+```Go
+return reconcile.Result{Requeue: true}, nil
+```
+
+Example:
+
+In the [`memcached_controller.go`][memcached_controller] example below a `reconcile.Result{Requeue: true}, nil` is used because a new resources is being created and as such there is the 
+potential that further processing is required. 
+Thus, the reconcile loop needs to trigger a requeue but there is no error associated with this requeue.
+As a result the request is requeued for another try.
+```Go
+// Define a new deployment
+dep := r.deploymentForMemcached(memcached)
+...
+
+// Deployment created successfully - return and requeue
+return reconcile.Result{Requeue: true}, nil
+```
+
+#### 3. Without an error and no need to requeue the request:
+
+In some situations, such as when the primary resource has been deleted, there is no need to
+requeue the request for another attempt
+
+Usage
+```Go
+return reconcile.Result{}, nil
+```
+
+Example:
+
+In the [`memcached_controller.go`][memcached_controller] example below a `reconcile.Result{}, nil` is used because the Memcached resource was not found, and no further processing is required.  
```

Comment:
```suggestion
In the example below a `reconcile.Result{}, nil` is used because the Memcached resource was not found, and no further processing is required.  
```

Reply:
Fixed